app/models/labware.rb

```diff
@@ -29,7 +29,16 @@ class Labware < Asset
   has_many :requests_as_target, through: :receptacles
   has_many :transfer_requests_as_source, through: :receptacles
   has_many :transfer_requests_as_target, through: :receptacles
+
+  # Submissions in progress for the labware. Found by looking at those associated
+  # with the transfer requests into the receptacles. Be a little cautious using this,
+  # as it will not handle cross-submission pools, but is better for historical
+  # data.
```

Comment:
Is it necessarily submissions 'in progress'? Is it filtering by state of the transfer request somewhere?

Why does it not include cross-submission pools? (Think I might need a diagram here!!)

Reply:
I'll need to revisit the working here then, although I'm not quite sure how to make it clear, as its all quite complicated.
So, in this case in-progress submissions is intending to mean, those you're in the act of processing currently, instead of those that are associated with requests out of the plate. In this particular situation the information gets pulled from the submission associated with transfer request.
In cross submission pools the transfer_requests don't link submission, as they are associated with multiple submissions. At this point we added the direct link from aliquot to submission. Data wasn't back-filled due to the sheer size and complexity of the task, although I am increasingly regretting that decision.

Reply:
Hmm, we have in_progress_submissions below, which is the same as this but with 'distinct' (probably what confused you below. I'll flag this one as deprecated as its really confusing, and tidy up the comments elsewhere

Reply:
OK thanks for explaining! 

So this returns submissions that are linked to any transfer requests into the labware's receptacles?
Still not sure why that means they are 'in progress' / 'you are in the act of processing currently' - wouldn't it pull submissions that were all done and finished with if you called it on an old labware?

Wonder if this area is worth a Confluence page (maybe there's a relevant one already?) because it's quite complicated, as you say! Happy to help with that, if I can.

Reply:
More inclined to suggest an md file in the sequencescape repo itself, can get yard to link the various sections of it then as relevant, and slightly less likely to get horrendously out of date.

You're right that the association will also return submissions that were done and dusted. The name originally came out of a requirement in limber to distinguish between work that was in progress, compared to work that was yet to start. Then name then cascaded across into sequencescape, where out of context it doesn't make as much sense.

That said, its probably worth auditing what's in place, and whether any of it can be removed. A lot of the logic these were originally added to support has been moved off into Limber, so it might be possible to strip some of it out.